pallets/schemas/src/migrations.rs

```diff
@@ -0,0 +1,23 @@
+//! Migrations for the schemas pallet.
+//! Following migrations are required:
+//! - Adding settings to the Schema struct
+//! Note: Post migration, this file should be deleted.
+use crate::{types::Schema, *};
+use common_primitives::schema::SchemaSettings;
+use frame_support::pallet_prelude::Weight;
+
+pub fn migrate_schemas_with_additional_settings<T: Config>() -> Weight {
+	let mut weight: Weight = Weight::zero();
+	//translate_values of Schemas storage add settings: SchemaSettings::all_disabled()
+	//to all schemas
+	<Schemas<T>>::translate_values(|old_schema: Schema<T::SchemaModelMaxBytesBoundedVecLimit>| {
+		weight = weight.saturating_add(T::DbWeight::get().reads_writes(1, 1));
+		Some(Schema {
+			model_type: old_schema.model_type,
+			model: old_schema.model,
+			payload_location: old_schema.payload_location,
+			settings: SchemaSettings::all_disabled(),
+		})
+	});
+	weight
+}
```

Comment:
@aramikm  so since we are simply translating values, this is some places where parity simply uses on_runtime_upgrade hook to run migrations

Reply:
Is there any possibly that a runtime migration hook might get triggered more than 1 time?

Reply:
For safety reasons, a migration hook should be designed to be idempotent even if you expect to remove it before the next runtime upgrade.  There should be some check like for if it's OldSchema or not before calling `translate_values`.

Reply:
@shannonwells I believe this `|old_schema: OldSchema<T::SchemaModelMaxBytesBoundedVecLimit>|` cast will prevent translation of any schemas that dont fit the OldSchema model since any new schemas wont decode to OldSchema type

Or you mean its safe to` iter -> check -> update?`

Reply:
I think she means checking even before iteration. One thing I've seen some pallets do it that they have a separate storage to keep track of current version of migration and update it after the migration is complete. So if the process runs again it would check with that version and not run if it's the latest version. There might be other ways also to achieve this.

Reply:
I see, yes I remember seeing that around, will redo this part